### ISSUE TITLE
"cabal build all"  fails 

### ISSUE THREAD

**Issue by SamirBitzees** [input-output-hk/cardano-db-sync#1517]
**OS**
Your OS: 
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04

Cable version :
cabal-install version 3.6.2.0
compiled using version 3.6.2.0 of the Cabal library 

ghc version:
The Glorious Glasgow Haskell Compilation System, version 8.10.7

**Additional context**
I tried these few steps also

1)cabal clean
2)cable build all 
3)sudo apt-get install libsodium-dev
4)sudo apt install libscrypt-dev

**Problem Report**
Installing   regex-pcre-builtin-0.95.2.3.8.44 (lib)
Completed    regex-pcre-builtin-0.95.2.3.8.44 (lib)
Installing   Cabal-syntax-3.10.1.0 (lib)
Completed    Cabal-syntax-3.10.1.0 (lib)

Failed to build scrypt-0.5.0. The failure occurred during the configure step.
Build log (
/home/ubuntu/.cabal/logs/ghc-8.10.7/scrypt-0.5.0-9dc7f5056f17df13813f95e91bb667ceed7dc3e60a8b91da16ab03a3c4c8a06b.log
):
Configuring library for scrypt-0.5.0..
cabal-3.6.2.0: Missing dependency on a foreign library:
* Missing (or bad) header file: crypto_scrypt.h
This problem can usually be solved by installing the system package that
provides this library (you may need the "-dev" version). If the library is
already installed but in a non-standard location then you can use the flags
--extra-include-dirs= and --extra-lib-dirs= to specify where it is.If the
library file does exist, it may contain errors that are caught by the C
compiler at the preprocessing stage. In this case you can re-run configure
with the verbosity flag -v3 to see the error messages.
If the header file does exist, it may contain errors that are caught by the C
compiler at the preprocessing stage. In this case you can re-run configure
with the verbosity flag -v3 to see the error messages.

cabal: Failed to build scrypt-0.5.0 (which is required by test:unit from
cardano-wallet-2023.7.18, exe:mock-token-metadata-server from
cardano-wallet-2023.7.18 and others). See the build log above for details.

**Comment by erikd** [comment]
I have just built git HEAD with `ghc-8.10.7` without seeing that issue. Also,there is no `crypto_scrypt.h` header file on my system. Not sure what this is all about?

**Comment by rdlrt** [comment]
Based on error message - it seems he is compiling cardano-wallet anyways, should be closed here...

He is copy pasting message without proper context across all repos and dev channels without looking at replies to follow threads (not the first time)

**Comment by erikd** [comment]
@SamirBitzees What package are you compiling?

**Comment by SamirBitzees** [comment]
> @SamirBitzees What package/repo are you compiling?

"I am attempting to compile the cardano-wallet package from the cardano-wallet repository. Specifically, I am working with the version labeled as 2023.7.18."

**Comment by SamirBitzees** [comment]
> I have just built git HEAD with `ghc-8.10.7` without seeing that issue. Also,there is no `crypto_scrypt.h` header file on my system. Not sure what this is all about?

"It's interesting to note that you didn't encounter this issue, and I appreciate your input. In my case, I'm using ghc-8.10.7 on Ubuntu 22.04.3 LTS, and I did indeed encounter the issue with the missing crypto_scrypt.h header file during the build process. It's possible that this issue is specific to my environment or configuration.

I am using this development guide document "[https://developers.cardano.org/docs/get-started/installing-cardano-wallet]

To provide some additional context, I performed the following steps before attempting the build:

cabal clean
cable build all
Installed system packages libsodium-dev and libscrypt-dev.
Despite these steps, the issue persisted. Any insights or suggestions on how to resolve this would be greatly appreciated."
